logger/logger_types.go

```diff
@@ -2,34 +2,90 @@ package logger
 
 import (
 	"io"
-	"log"
-	"os"
+	"sync"
 )
 
-// Logging struct that holds all user configurable options for the logger
-type Logging struct {
-	Enabled              *bool  `json:"enabled,omitempty"`
-	File                 string `json:"file"`
-	ColourOutput         bool   `json:"colour"`
-	ColourOutputOverride bool   `json:"colourOverride,omitempty"`
-	Level                string `json:"level"`
-	Rotate               bool   `json:"rotate"`
+const timestampFormat = " 02/01/2006 15:04:05 "
+
+const spacer = "|"
+
+type Config struct {
+	Enabled          *bool            `json:"enabled"`
+	AdvancedSettings advancedSettings `json:"advancedSettings"`
+	SubLoggers       []SubLoggers     `json:"subloggers"`
 }
 
-var (
-	debugLogger *log.Logger
-	infoLogger  *log.Logger
-	warnLogger  *log.Logger
-	errorLogger *log.Logger
-	fatalLogger *log.Logger
+type headers struct {
+	Info  string `json:"info"`
+	Warn  string `json:"warn"`
+	Debug string `json:"debug"`
+	Error string `json:"error"`
+}
+
+type advancedSettings struct {
+	Spacer          string  `json:"spacer"`
+	TimeStampFormat string  `json:"timeStampFormat"`
+	Headers         headers `json:"headers"`
+}
 
-	logFileHandle *os.File
+type SubLoggers struct {
+	Name   string `json:"name"`
+	Level  string `json:"level"`
+	Output string `json:"output"`
+}
 
-	logOutput io.Writer
+type Logger struct {
+	Timestamp                                        string
+	InfoHeader, ErrorHeader, DebugHeader, WarnHeader string
+	Spacer                                           string
+}
+
+type subLogger struct {
+	Info, Debug, Warn, Error bool
+	output                   io.Writer
+}
 
-	// LogPath location to store logs in
+type LogEvent struct {
+	data   []byte
+	output io.Writer
+}
+
+type multiWriter struct {
+	writers []io.Writer
+	mu      sync.Mutex
+}
+
+var (
+	logger           = &Logger{}
+	GlobalLogConfig  = &Config{}
+	subSystemLoggers = map[string]subLogger{}
+	eventPool        = &sync.Pool{
+		New: func() interface{} {
+			return &LogEvent{
+				data: make([]byte, 0, 80),
+			}
+		},
+	}
 	LogPath string
+)
+
+const (
+	LogGlobal = "log"
+
+	SubSystemConnMgr = "connection"
+	SubSystemCommMgr = "communications"
+	SubSystemConfMgr = "config"
+	SubSystemOrdrMgr = "order"
+	SubSystemPortMgr = "portfolio"
+	SubSystemSyncMgr = "syncer"
+	SubSystemTimeMgr = "timekeeper"
+	SubSystemWsocMgr = "websocket"
+	SubSystemEvntMgr = "event"
```

Comment:
Another minor thought is that this could be a custom type. You already have some code to protect from custom subsystems from being passed in, but if someone is looking to log and you see a `string` is passed in, one could easily conclude that they could input anything. The lack of any comments on the[ `func Info`](https://github.com/thrasher-/gocryptotrader/pull/319/files#diff-5c91e5a763e453613ac4f04d193662fcR5) etc allow for this too. Only at compile time will they discover that their log is not allowed and instead just says `log`. 

By having a type, a user can know that there is a strict definition of the kinds of subsystems that can be passed to it and then append any when new subsystems are developed.